### PR TITLE
SUS-2474: Improve efficiency of Wall CommentsIndex lookups

### PR DESCRIPTION
**Pull request by mszabo-wikia**
* Swap order of `TemplateClassification` hook conditions - quickly checking namespace is much faster than triggering whole `userCan` hook execution.
* Remove useless fallback to master in `CommentsIndex` - it never helped
* Tackle a few cases where we passed non-existing title to `WallMessage`, resulting in `commentId = 0`:
-- redirect to Main Page if someone tries to access invalid thread - as the thread does not exist, trying to look up its message wall will always fail
-- get wall owner name from title instead of going through `CommentsIndex`

https://wikia-inc.atlassian.net/browse/SUS-2474

### CODE REVIEW

File: extensions/wikia/Wall/WallTabsRenderer.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			array_filter( $tabClasses, function ( $class ) {
+				return $class !== 'new';
+			} );
+	}
+
+	public function renderWallContentActions( array &$contentActions ) {
+		$messageTitle = $this->getMessageTitle();
+		$wallTitleText = $messageTitle->getBaseText();
+
+		$owningUser = User::newFromName( $wallTitleText, $validateUserName = false );
```
Comment (WikiaTech) [inline extensions/wikia/Wall/WallTabsRenderer.php:50]: MAJOR Remove this unused "$validateUserName" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/TemplateClassification/Permissions.class.php [hunk 2]
```diff
@@ -15,8 +15,8 @@ class Permissions {
 	 * @return bool
 	 */
 	public function shouldDisplayEntryPoint( \User $user, \Title $title ) {
-		return $this->userCanChangeType( $user, $title )
-		       && $title->inNamespace( NS_TEMPLATE );
+		return $title->inNamespace( NS_TEMPLATE ) &&
+			   $this->userCanChangeType( $user, $title );
```
Comment (macbre) [inline extensions/wikia/TemplateClassification/Permissions.class.php:19]: :+1:

**Review by macbre — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

**Comment by rybmat** [comment]
merge dev, tests should pass now

**Pull request merged**